src/app/shared/search-box/query-item/query-items-converter.ts

```diff
@@ -76,10 +76,15 @@ export class QueryItemsConverter {
 
     if (query.linkTypeIds && query.linkTypeIds.length > 0) {
       return this.linkTypeService.getLinkTypes({linkTypeIds: query.linkTypeIds}).pipe(
-        map((linkTypes: LinkType[]) => linkTypes.map(linkType => linkType.collectionCodes)),
-        map(([codes]: string[][]) => codes.filter(code => !collectionCodes.includes(code))),
-        map(codes => codes.concat(collectionCodes)),
-        switchMap(codes => this.searchService.searchCollections({collectionCodes: codes}))
+        map((linkTypes: LinkType[]) => {
+          const set = new Set<string>();
+          linkTypes.forEach(linkType => {
+            set.add(linkType.collectionIds[0]);
+            set.add(linkType.collectionIds[1]);
```

Comment:
It is not the best idea to use `Set` here since you can even create link within the same collection.

Reply:
But we want to fetch each collection just once.